kuma/wiki/models.py

```diff
@@ -925,6 +931,11 @@ def save(self, *args, **kwargs):
         # Delete any cached last-modified timestamp.
         self.fill_last_modified_cache()
 
+        # Set the shortened URL (e.g. bit.ly). Note: We do this only for new
+        # documents that aren't templates or redirects.
+        if is_new and not self.is_template and not self.is_redirect:
```

Comment:
Please put this into the `on_document_save` signal handler in the `wiki/apps.py` and check for the `created` flag passed into the signal handler. That's the canonical place for such things now